### DISCUSSION TITLE
Cross compiling wheel for armv5te

### DISCUSSION THREAD

**Discussion by lscorcia** [PyO3/maturin#2405] [Q&A] [answered]
Hi, I'm trying to cross compile a wheel for package uv using maturin. I'm on an Ubuntu machine and I'm trying to build for the armv5te architecture. uv contains a rust module that I have been able to cross compile successfully using:

`CARGO_TARGET_ARMV5TE_UNKNOWN_LINUX_GNUEABI_LINKER="/usr/bin/arm-linux-gnueabi-gcc" cargo build --release --target armv5te-unknown-linux-gnueabi`

However, when running maturin I do get the wrong architecture out:

`CARGO_TARGET_ARMV5TE_UNKNOWN_LINUX_GNUEABI_LINKER="/usr/bin/arm-linux-gnueabi-gcc" maturin build --release --target armv5te-unknown-linux-gnueabi --compatibility linux`

Completes successfully, but builds the following:

`uv-0.5.11-py3-none-linux_armv7l.whl`

Installing this wheel on the machine reports this error:

ERROR: uv-0.5.11-py3-none-linux_armv7l.whl is not a supported wheel on this platform.

No additional info, but wheels built on the machine itself usually end with `_armv5tel.whl`.
Am I missing some other flag or parameter?

[sysconfig.txt](https://github.com/user-attachments/files/18222612/sysconfig.txt)

**Comment by messense** [comment, marked as answer]
Add support for armv5te in https://github.com/PyO3/maturin/pull/2409, can you confirm some information on the machine?

1. output of `uname -m`
2. output of `pip debug --verbose`

**Comment by lscorcia** [comment]
$ uname -m
armv5tel

$ pip debug --verbose
pip version: pip 24.0 from /srv/homeassistant/lib/python3.12/site-packages/pip (python 3.12)
sys.version: 3.12.3 (main, Dec 11 2024, 09:13:12) [GCC 12.2.0]
sys.executable: /srv/homeassistant/bin/python3.12
sys.getdefaultencoding: utf-8
sys.getfilesystemencoding: utf-8
locale.getpreferredencoding: utf-8
sys.platform: linux
sys.implementation:
  name: cpython
'cert' config value: Not specified
REQUESTS_CA_BUNDLE: None
CURL_CA_BUNDLE: None
pip._vendor.certifi.where(): /srv/homeassistant/lib/python3.12/site-packages/pip/_vendor/certifi/cacert.pem
pip._vendor.DEBUNDLED: False
vendored library versions:
  CacheControl==0.13.1
  colorama==0.4.6
  distlib==0.3.8
  distro==1.8.0
  msgpack==1.0.5
  packaging==21.3
  platformdirs==3.8.1
  pyparsing==3.1.0
  pyproject-hooks==1.0.0
  requests==2.31.0
  certifi==2023.07.22
  chardet==5.1.0
  idna==3.4
  urllib3==1.26.17
  rich==13.4.2 (Unable to locate actual module version, using vendor.txt specified version)
  pygments==2.15.1
  typing_extensions==4.7.1 (Unable to locate actual module version, using vendor.txt specified version)
  resolvelib==1.0.1
  setuptools==68.0.0 (Unable to locate actual module version, using vendor.txt specified version)
  six==1.16.0
  tenacity==8.2.2 (Unable to locate actual module version, using vendor.txt specified version)
  tomli==2.0.1
  truststore==0.8.0
  webencodings==0.5.1 (Unable to locate actual module version, using vendor.txt specified version)
Compatible tags: 42
  cp312-cp312-linux_armv5tel
  cp312-abi3-linux_armv5tel
  cp312-none-linux_armv5tel
  cp311-abi3-linux_armv5tel
  cp310-abi3-linux_armv5tel
  cp39-abi3-linux_armv5tel
  cp38-abi3-linux_armv5tel
  cp37-abi3-linux_armv5tel
  cp36-abi3-linux_armv5tel
  cp35-abi3-linux_armv5tel
  cp34-abi3-linux_armv5tel
  cp33-abi3-linux_armv5tel
  cp32-abi3-linux_armv5tel
  py312-none-linux_armv5tel
  py3-none-linux_armv5tel
  py311-none-linux_armv5tel
  py310-none-linux_armv5tel
  py39-none-linux_armv5tel
  py38-none-linux_armv5tel
  py37-none-linux_armv5tel
  py36-none-linux_armv5tel
  py35-none-linux_armv5tel
  py34-none-linux_armv5tel
  py33-none-linux_armv5tel
  py32-none-linux_armv5tel
  py31-none-linux_armv5tel
  py30-none-linux_armv5tel
  cp312-none-any
  py312-none-any
  py3-none-any
  py311-none-any
  py310-none-any
  py39-none-any
  py38-none-any
  py37-none-any
  py36-none-any
  py35-none-any
  py34-none-any
  py33-none-any
  py32-none-any
  py31-none-any
  py30-none-any
